client/app/bundles/course/assessment/submissions/pages/SubmissionsIndex/index.tsx

```diff
@@ -0,0 +1,257 @@
+/* eslint-disable no-nested-ternary */
+import { defineMessages, injectIntl, WrappedComponentProps } from 'react-intl';
+import { FC, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, AppState } from 'types/store';
+
+import { toast } from 'react-toastify';
+
+import LoadingIndicator from 'lib/components/LoadingIndicator';
+import { Badge, BadgeProps, styled, Tab, Tabs } from '@mui/material';
+
+import PageHeader from 'lib/components/pages/PageHeader';
+import {
+  getAllSubmissionMiniEntities,
+  getFilter,
+  getIsGamified,
+  getSubmissionCount,
+  getSubmissionPermissions,
+  getTabs,
+} from '../../selectors';
+import {
+  fetchAllStudentsPendingSubmissions,
+  fetchCategorySubmissions,
+  fetchMyStudentsPendingSubmissions,
+  fetchSubmissions,
+} from '../../operations';
+import SubmissionsTable from '../../components/tables/SubmissionsTable';
+import SubmissionFilter from '../../components/misc/SubmissionFilter';
+
+interface Props extends WrappedComponentProps {}
+
+const translations = defineMessages({
+  header: {
+    id: 'course.assessments.submissions.header',
+    defaultMessage: 'Submissions',
+  },
+  fetchSubmissionsFailure: {
+    id: 'course.assessments.submissions.fetchSubmissionsFailure',
+    defaultMessage: 'Failed to fetch submissions',
+  },
+  allStudentsPending: {
+    id: 'course.assessments.submissions.allStudentsPending',
+    defaultMessage: 'All Pending Submissions',
+  },
+  myStudentsPending: {
+    id: 'course.assessments.submissions.myStudentsPending',
+    defaultMessage: 'My Students Pending',
+  },
+  noSubmissionsMessage: {
+    id: 'course.assessments.submissions.noSubmissionsMessage',
+    defaultMessage: 'There are no submissions',
+  },
+});
+
+const CustomBadge = styled(Badge)<BadgeProps>(({ theme }) => ({
+  '& .MuiBadge-badge': {
+    right: -5,
+    top: 0,
+    border: `2px solid ${theme.palette.background.paper}`,
+    padding: '0 4px',
+  },
+}));
+
+const SubmissionsIndex: FC<Props> = (props) => {
+  const { intl } = props;
+
+  const ROWS_PER_PAGE = 25;
+  const [pageNum, setPageNum] = useState(1);
+
+  // Selectors
+  const isGamified = useSelector((state: AppState) => getIsGamified(state));
+  const submissionCount = useSelector((state: AppState) =>
+    getSubmissionCount(state),
+  );
+  const submissions = useSelector((state: AppState) =>
+    getAllSubmissionMiniEntities(state),
+  );
+  const tabs = useSelector((state: AppState) => getTabs(state));
+  const filter = useSelector((state: AppState) => getFilter(state));
+  const submissionPermissions = useSelector((state: AppState) =>
+    getSubmissionPermissions(state),
+  );
+
+  // For tab logic and control
+  const [tabValue, setTabValue] = useState(2);
```

Comment:
Could we set a proper name for the tabValue instead of using magic number?

Reply:
If i dont set it with a number the state has a very complicated update that I cant really solve, I'll put in comments explaining what the numbers mean